### ISSUE TITLE
Batch-normalize projective to affine conversions

### ISSUE THREAD

**Issue by TalDerei** [tachyon-zcash/ragu#545]
We should use batch normalize (montgomery's trick) to convert N projective points (from commits) to affine with 1 inversion + 3(N-1) multiplications instead of N inversions.
